### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:

- Mix format module config
- Use common source url
- Show source reference by tagged version
- Use shorter module name
- Fix incorrect changelog link
- Badges and more badges!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -2,55 +2,68 @@ defmodule PropCheck.Mixfile do
   @moduledoc "Buildfile for PropCheck"
   use Mix.Project
 
+  @source_url "https://github.com/alfert/propcheck"
+  @version "1.3.1-dev"
```
Comment (alfert) [inline mix.exs:6]: this requires also fixes in release.sh ...

File: README.md [hunk 2]
```diff
@@ -18,24 +22,24 @@ defp deps do
...
-From PropcCheck 1.3.0 onwards, we require at least Elixir 1.7 since in Elixir 1.11 function `get_stracktrace()` is deprecated. However, we 
+From PropcCheck 1.3.0 onwards, we require at least Elixir 1.7 since in Elixir 1.11 function `get_stracktrace()` is deprecated. However, we
 only support OTP 22, since `:proper` does not yet support OTP 23 in a released version. Hopefully, this will change soon! Running under
-OTP 23 results in no stacktraces - which is not helpful for identifying bugs. 
+OTP 23 results in no stacktraces - which is not helpful for identifying bugs.
 
 ## Changes
 
-Relevant changes are documented in the [Changelog](changelog.html), on [GitHub
+Relevant changes are documented in the [Changelog](https://hexdocs.pm/propcheck/changelog.html), on [GitHub
```
Comment (alfert) [inline README.md:31]: In which scenario does this not work? I added both links to provide proper links for github, Dash and hexdocs - at least this was the plan

**Comment by alfert** [comment]
Thanks for your PR! 💚 

I have go a few questions though ...

File: README.md (hunk 2 above)
Comment (kianmeng) [reply README.md:31]: In the current https://github.com/alfert/propcheck/blob/master/README.md, the `Changelog` link shows 404, which points to https://github.com/alfert/propcheck/blob/master/changelog.html

File: mix.exs (hunk 1 above)
Comment (kianmeng) [reply mix.exs:6]: > this requires also fixes in release.sh ...

Done. Let me know if you understand your `release.sh` script correctly.

Comment (alfert) [reply mix.exs:6]: Sorry for my late response, too much to do in real life. 

What I meant are the `sed` commands the replace the value of `version: ` but must now replace `@version `. If you feel uncomfortable with that change, don't mind. I can do it later.

**Comment by alfert** [comment]
Please do a rebase, that I can merge your PR!

**Comment by alfert** [comment]
I close your PR, since I cherrypicked most of the changes into the current PR #186. Thanks for contributing!

**Comment by kianmeng** [comment]
@alfert Sorry, I miss the notification. Glad I can help.

**Pull request closed without merging**